scripts/directtools/__init__.py

```diff
@@ -0,0 +1,450 @@
+from __future__ import (absolute_import, division, print_function)
+
+import collections
+from mantid import mtd
+from mantid.simpleapi import DeleteWorkspace, LineProfile, OneMinusExponentialCor, Transpose
+import matplotlib
+from matplotlib import pyplot
+import numpy
+from scipy import constants
+import time
+
+
+def _chooseMarker(markers, index):
+    """Pick a marker from markers and return next cyclic index."""
+    if index >= len(markers):
+        index = 0
+    marker = markers[index]
+    index += 1
+    return marker, index
+
+
+def _clearlatex(s):
+    """Return a string from which LaTeX formatting has been removed."""
+    s = s.translate(None, '$\\^{}')
+    s = s.replace('$', '')
+    s = s.replace('\\', '')
+    s = s.replace('^', '')
+    s = s.replace('{', '')
+    s = s.replace('}', '')
+    return s
+
+
+def _finalizeprofileE(axes):
+    """Set axes for const E axes."""
+    axes.set_xlim(xmin=0.)
+    axes.set_xlabel('$Q$ (\\AA$^{-1}$)')
+    axes.set_ylim(0.)
+    xMin, xMax = axes.get_xlim()
+    print('Auto Q-range: {}...{} \xc5-1'.format(xMin, xMax))
+
+
+def _finalizeprofileQ(workspaces, axes):
+    """Set axes for const Q axes."""
+    workspaces = _normwslist(workspaces)
+    axes.set_xlim(xmin=-10.)
```

Comment:
Why -10?

Reply:
The axis limits are 'sane defaults' requested by the instrument scientists to cover the most common first looks at the *S(Q, E)* data. One can adjust the limits later using `matplotlib` functions, if needed.

Reply:
Those are save values for a particular instrument. I am thinking that minimum energy of -10 has different meaning for CNCS with Ei=1.5meV versus SEQUOIA with Ei=1500 meV. Your minimum should probably be a percentage of Ei.

Reply:
This could be a nice future improvement, indeed. I'm planning to develop this module further and make it less ILL specific, anyhow.